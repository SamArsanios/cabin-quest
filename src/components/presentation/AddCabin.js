import React, { Component } from 'react';
import { Button, Modal } from 'react-bootstrap';
import PropTypes from 'prop-types';
import AddCabinForm from './AddCabinForm';
import ImageUploader from './ImageUploader';

class AddCabin extends Component {
  constructor(props) {
    super(props);

    this.state = {
      show: false,
    };
  }

  render() {
    const handleShow = () => {
      this.setState({
        show: true,
      });
    };

    const handleClose = () => {
      this.setState({
        show: false,
      });
    };
    const { cabin, status } = this.props;
    const { show } = this.state;

    return (
      <div>
        <Button
          variant="transparent"
          className="btn hero-btn cus-btn w-100"
          onClick={handleShow}
        >
          {`${status}  Cabin`}
        </Button>

        <Modal show={show} onHide={handleClose}>
          <Modal.Header closeButton>
            <Modal.Title>Add a new Cabin</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <ImageUploader status={status} />
            <AddCabinForm close={handleClose} cabin={cabin} status={status} />
          </Modal.Body>
          <Modal.Footer />
        </Modal>
      </div>
    );
  }
}
AddCabin.propTypes = {
  cabin: PropTypes.any,
  status: PropTypes.any,
};

export default AddCabin;
