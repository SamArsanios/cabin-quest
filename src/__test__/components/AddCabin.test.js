// import { render, screen } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import AddCabin from '../../components/presentation/AddCabin';

describe('AddCabin', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <AddCabin />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

