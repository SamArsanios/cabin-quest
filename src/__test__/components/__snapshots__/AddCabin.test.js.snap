// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCabin renders correctly 1`] = `
<div>
  <button
    className="btn hero-btn cus-btn w-100 btn btn-transparent"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    undefined  Cabin
  </button>
</div>
`;
